controllers/applicationController.go

```diff
@@ -110,6 +110,48 @@ func CORSOptionsHandler(w http.ResponseWriter, r *http.Request, next http.Handle
 	next(w, r)
 }
 
+var RedisConn redis.Conn
+
+func CheckCache() func(http.ResponseWriter, *http.Request, http.HandlerFunc) {
+	return func(w http.ResponseWriter, r *http.Request, next http.HandlerFunc) {
+		url := r.URL.Path + "?" + r.URL.RawQuery
+		cr, err := RedisConn.Do("GET", url)
+		if err != nil {
+			log.Fatal("Connection failure to Redis!")
+			// Might want to rethink this
+		}
+		if cr == nil {
+			//log.Printf("Cache miss: "+url)
+			next(w, r)
+			return
+		}
+		//log.Printf("Cache HIT: "+url)
+		sendJSONResponseNoCache(w, cr.([]byte))
+	}
+}
+
+func sendJSONResponseNoCache(w http.ResponseWriter, payload []byte) {
+	w.Header().Set("Content-Type", "application/json")
+	w.WriteHeader(http.StatusOK)
+	w.Write(payload)
+}
+
+func SendJSONResponse(w http.ResponseWriter, r *http.Request, payload []byte) {
+	sendJSONResponseNoCache(w, payload)
+
+	url := r.URL.Path + "?" + r.URL.RawQuery
+	resp, err := RedisConn.Do("SET", url, payload)
+	if err != nil {
+		log.Fatal("Connection failure to Redis!")
+		// Might want to rethink this
```

Comment:
Definitely. We do not want a failure to cache to cause the application itself to crash.